projects/packages/videopress/src/client/admin/components/video-upload-area/index.tsx

```diff
@@ -0,0 +1,110 @@
+/**
+ * External dependencies
+ */
+import { Button, useBreakpointMatch, Text } from '@automattic/jetpack-components';
+import { Spinner } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { Icon, cloudUpload } from '@wordpress/icons';
+import classnames from 'classnames';
+import { DragEvent, useCallback, useState, useRef, useEffect } from 'react';
+/**
+ * Internal dependencies
+ */
+import styles from './style.module.scss';
+import { VideoUploadAreaProps } from './types';
+import type React from 'react';
+
+/**
+ * Video Upload Area component
+ *
+ * @param {VideoUploadAreaProps} props - Component props.
+ * @returns {React.ReactNode} - VideoUploadArea react component.
+ */
+const VideoUploadArea: React.FC< VideoUploadAreaProps > = ( {
+	isLoading = false,
+	className,
+	onSelectFiles,
+} ) => {
+	const [ isSm ] = useBreakpointMatch( 'sm' );
+	const [ isDraggingOver, setIsDraggingOver ] = useState( false );
+	const inputRef = useRef( null );
+
+	const handleFileInputChangeEvent = useCallback( ( e: Event ) => {
+		const target = e.target as HTMLInputElement;
+		onSelectFiles( target.files );
+	} );
+
+	const handleClickEvent = useCallback( () => {
+		inputRef.current.click();
+	}, [] );
+
+	const handleDragOverEvent = useCallback( ( event: DragEvent< HTMLInputElement > ) => {
+		event.preventDefault();
+		setIsDraggingOver( true );
+	}, [] );
+
+	const handleDragLeaveEvent = useCallback( () => {
+		setIsDraggingOver( false );
+	}, [] );
+
+	const handleDropEvent = useCallback(
+		( event: DragEvent< HTMLInputElement > ) => {
+			event.preventDefault();
+			setIsDraggingOver( false );
+
+			onSelectFiles( event.dataTransfer.files );
+		},
+		[ onSelectFiles ]
+	);
+
+	useEffect( () => {
+		if ( isLoading ) {
+			setIsDraggingOver( false );
+		}
+	}, [ isLoading ] );
+
+	return (
+		<div
+			className={ classnames( styles.wrapper, className, {
+				[ styles.small ]: isSm,
+				[ styles.hover ]: isDraggingOver,
+			} ) }
+			{ ...( ! isLoading && {
+				onDrop: handleDropEvent,
+				onDragOver: handleDragOverEvent,
+				onDragLeave: handleDragLeaveEvent,
+			} ) }
+		>
+			{ isLoading ? (
+				<>
+					<Spinner className={ classnames( styles.loader ) } />
+					<Text variant="title-small">{ __( 'Uploading', 'jetpack-videopress-pkg' ) }</Text>
+				</>
```

Comment:
Based on Noam comments here: p1660737908278569/1660684867.493439-slack-C03TA48NSUX

and Figma:

![image](https://user-images.githubusercontent.com/1663717/185253494-bd280f00-1a64-4f38-8472-df32d791d9c3.png)

The `Loading` should not be implemented here. Right?

It will be implemented by the parent.

Reply:
@renatoagds I'm going to accept this PR since it's getting a blocker. It implements part of the compiling system supporting TS. Let's address this issue in a follow-up PR.